packages/x-date-pickers-pro/src/hooks/useMultiInputRangeField/useMultiInputRangeField.ts

```diff
@@ -0,0 +1,180 @@
+'use client';
+import useEventCallback from '@mui/utils/useEventCallback';
+import {
+  FieldChangeHandler,
+  FieldChangeHandlerContext,
+  PickerManagerError,
+  PickerManagerFieldInternalProps,
+  PickerRangeValue,
+  PickerValue,
+  useControlledValueWithTimezone,
+  useFieldInternalPropsWithDefaults,
+} from '@mui/x-date-pickers/internals';
+import { useValidation } from '@mui/x-date-pickers/validation';
+import { useMultiInputRangeFieldTextFieldProps } from './useMultiInputRangeFieldTextFieldProps';
+import { useMultiInputRangeFieldSelectedSections } from './useMultiInputRangeFieldSelectedSections';
+import { PickerAnyRangeManager } from '../../internals/models/managers';
+
+/**
+ * Basic example:
+ *
+ * ```tsx
+ * import Box from '@mui/material/Box';
+ * import { useSplitFieldProps } from '@mui/x-date-pickers/hooks';
+ * import { PickersTextField } from '@mui/x-date-pickers/PickersTextField';
+ * import { useDateRangeManager } from '@mui/x-date-pickers-pro/managers';
+ *
+ * function MultiInputField(props) {
+ *   const manager = useDateRangeManager();
+ *   const { internalProps, forwardedProps } = useSplitFieldProps(props, 'date');
+ *   const { startDate, endDate } = useMultiInputRangeField({
+ *     manager,
+ *     internalProps,
+ *     startForwardedProps: forwardedProps,
+ *     endForwardedProps: forwardedProps,
+ *   });
+ *
+ *   return (
+ *     <Box {...forwardedProps}>
+ *       <PickersTextField {...startDate} />
+ *       <span>{' – '}</span>
+ *       <PickersTextField {...endDate} />
+ *     </Box>
+ *   );
+ * }
+ * ```
+ *
+ * @param {UseMultiInputRangeFieldParameters<TManager, TForwardedProps>} parameters The parameters of the hook.
+ * @param {TManager} parameters.manager The manager of the field.
+ * @param {PickerManagerFieldInternalProps<TManager>} parameters.internalProps The internal props of the field.
+ * @param {TForwardedProps} parameters.startForwardedProps The forwarded props of the start field.
+ * @param {TForwardedProps} parameters.endForwardedProps The forwarded props of the end field.
+ * @returns {{ startDate: Omit<UseFieldResponse<TEnableAccessibleFieldDOMStructure, TForwardedProps>, 'clearable' | 'onClear'>, endDate: Omit<UseFieldResponse<TEnableAccessibleFieldDOMStructure, TForwardedProps>, 'clearable' | 'onClear'> }} The props to pass to the start and the end components.
```

Comment:
Nitpick: I fear that we might forget to update such dynamic types comments. 🙈

Reply:
True, I'll create a clean internal type for it.

Reply:
Done